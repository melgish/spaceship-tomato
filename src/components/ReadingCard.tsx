import type { Reading } from '@/actions/findReadings'
import RemoveReadingButton from './RemoveReadingButton'

type Props = Readonly<Reading>

export default function ReadingCard(row: Props) {
  const plant = row.bucket.plant
  return (
    <div className="kv-card">
      <div>Created</div>
      <div>
        {row.createdAt.toLocaleString()} <RemoveReadingButton row={row} />
      </div>
      <div>Bucket</div>
      <div>{row.bucket.name}</div>
      <div>Plant</div>
      <div>{plant?.name}</div>
      <div>PH</div>
      <div>{row.ph.toFixed(1)}</div>
      <div>TDS</div>
      <div>{row.tds.toFixed(0)}</div>
      <div>EC</div>
      <div>{row.ec.toFixed(0)}</div>
      <div />
    </div>
  )
}
