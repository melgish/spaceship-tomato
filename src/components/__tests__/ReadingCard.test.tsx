import { expect, it } from 'vitest'
import { render } from '@testing-library/react'
import ReadingCard from '../ReadingCard'

// A typical reading.
const data = {
  id: 1,
  bucket: {
    id: 1,
    name: 'Test Bucket',
    plant: {
      id: 1,
      name: 'Test Plant',
      minPH: 6,
      maxPH: 7,
    },
    plantId: 1,
  },
  bucketId: 1,
  createdAt: new Date(Date.UTC(2024, 5, 13, 6, 12, 34, 45)),
  ec: 1234,
  ph: 6.5,
  tds: 805,
}

it('should display the supplied reading', () => {
  const { container } = render(<ReadingCard {...data} />)

  expect(container).toMatchSnapshot()
})
