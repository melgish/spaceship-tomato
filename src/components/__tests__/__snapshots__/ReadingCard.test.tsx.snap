// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should display the supplied reading 1`] = `
<div>
  <div
    class="kv-card"
  >
    <div>
      Created
    </div>
    <div>
      6/13/2024, 6:12:34 AM
       
      <button
        class="px-1 py-0 text-red-800 font-bold border-red-800 text-sm"
        type="button"
      >
        Remove
      </button>
    </div>
    <div>
      Bucket
    </div>
    <div>
      Test Bucket
    </div>
    <div>
      Plant
    </div>
    <div>
      Test Plant
    </div>
    <div>
      PH
    </div>
    <div>
      6.5
    </div>
    <div>
      EC
    </div>
    <div>
      1234
    </div>
    <div>
      TDS
    </div>
    <div>
      805
    </div>
    <div />
  </div>
</div>
`;
